apps/api/src/campaign/campaign.service.ts

```diff
@@ -116,6 +116,40 @@ export class CampaignService {
     return this.prisma.vault.findFirst({ where: { campaignId } })
   }
 
+  async getDonationsForCampaign(campaignId: string): Promise<Donation[]> {
+    const campaign = await this.prisma.campaign.findFirst({
+      where: { id: campaignId },
+      include: {
+        vaults: true,
+      },
+    })
+
+    if (campaign === null) {
+      Logger.warn('No campaign record with id: ' + campaign)
+      throw new NotFoundException('No campaign record with id: ' + campaign)
+    }
+
+    const whereVaultIds = campaign.vaults.map((vault) => {
+      return { targetVaultId: vault.id }
+    })
+
+    const donations = await this.prisma.donation.findMany({
+      where: {
+        OR: whereVaultIds,
+      },
+      include: {
+        person: { select: { firstName: true, lastName: true } },
+      },
+    })
+
+    donations.map((donation) => {
+      if (!donation.person)
+        donation.person = { firstName: 'anonymous', lastName: '' }
+    })
+
+    return donations
+  }
```

Comment:
the donations can be extracted from the database using a single query in this way:

```
const donations = await this.prisma.donation.findMany({
      where: { 
        status: 'succeeded',  //this is could be passed in the request too
        targetVault: {
          campaignId: { equals: campaignId}
        }
      },
      include: {
        person: { select: { firstName: true, lastName: true } },
      },
    })
```

and it generates empty array if there are no donations or no campaignId.

Example response:
`[
  {
    "id": "15cb9df5-ba2a-4d91-b409-3392a3d7348d",
    "type": "donation",
    "status": "succeeded",
    "provider": "none",
    "targetVaultId": "0ca7229e-af8b-4409-8c05-48e3318591ec",
    "extCustomerId": "",
    "extPaymentIntentId": "",
    "extPaymentMethodId": "",
    "createdAt": "2022-04-04T07:22:03.608Z",
    "updatedAt": "2022-04-04T07:30:14.787Z",
    "amount": 111,
    "currency": "BGN",
    "personId": null,
    "person": null
  }
]`